src/reuse/header.py

```diff
@@ -755,7 +767,7 @@ def run(args: Namespace, project: Project, out: IO[str] = sys.stdout) -> int:
             force_multi=args.multi_line,
         )
         if not args.skip_unrecognised:
```

Comment:
I would remove a lot of logic elsewhere and change this to

```suggestion
        if not args.skip_unrecognised or args.fallback_dot_license:
```

This reduces passing the parcel, and matches where `skip_unrecognised` is used.